compiler/rustc_mir/src/interpret/operand.rs

```diff
@@ -571,14 +571,16 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         }
     }
 
+    #[inline(never)]
     crate fn mir_const_to_op(
         &self,
-        val: &mir::ConstantKind<'tcx>,
+        val: mir::ConstantKind<'tcx>,
         layout: Option<TyAndLayout<'tcx>>,
     ) -> InterpResult<'tcx, OpTy<'tcx, M::PointerTag>> {
+        let val = self.subst_from_current_frame_and_normalize_erasing_regions(val);
         match val {
             mir::ConstantKind::Ty(ct) => self.const_to_op(ct, layout),
-            mir::ConstantKind::Val(val, ty) => self.const_val_to_op(*val, ty, None),
+            mir::ConstantKind::Val(val, ty) => self.const_val_to_op(val, ty, None),
```

Comment:
The perf regression in the CTFE tests must be due to these changes causing subtly different decisions in LLVM optimizations like inlining).

cc @tmiasko you've previously done some comparisons where you looked at differences in executions (or just symbols?) between two different `rustc`s. Are the instructions for doing so written up somewhere in the rustc guide or can you add them?

Reply:
Ok, I'm getting fairly far with

```
nm build/x86_64-unknown-linux-gnu/stage1/lib/librustc_driver-a150cfa449c4561a.so | grep eval_context | c++filt > symbols
```

Since this keeps coming up, I'll look into adding another page to the perf page for such information.

Reply:
To reproduce perf.rlo results locally under a profiler:

```
# Install and build rustc-perf
git clone https://github.com/rust-lang/rustc-perf
cd rustc-perf
cargo build --release --bin collector --bin rustc-fake

# Install rustup-toolchain-install-master and toolchains selected for comparison:
cargo install rustup-toolchain-install-master
A=d04c3aa8656f6588c87bafafb34d51239dab98bb
B=4151086b47b5fbb6f35855d9436a5a73a476f038
rustup-toolchain-install-master "$A" "$B"

# For example, to run ctfe-stress-4 check full under cachegrind.
./target/release/collector profile_local cachegrind $(rustup which rustc --toolchain "$A") a --builds Check --include ctfe-stress-4 --runs Full
./target/release/collector profile_local cachegrind $(rustup which rustc --toolchain "$B") b --builds Check --include ctfe-stress-4 --runs Full
# Output from a profiler is stored in "results" directory.
# See collector/README.md for other profilers and recommendations about using them.
# Edit collector/src/rustc-fake.rs to customize further.

# Continuing with the example, cg_diff between those two toolchains:
cg_diff results/cgout-{a,b}-ctfe-stress-4-Check-Full > results/cgdiff-ab
cg_annotate results/cgdiff-ab

# Cachegrind diff with an extra normalization of paths and function names:
cg_diff '--mod-filename=s#/rustc/[^/]*/##' '--mod-funcname=s/llvm[.].*/llvm/' results/cgout-{a,b}-ctfe-stress-4-Check-Full > results/cgdiff-ab
cg_annotate results/cgdiff-ab
```

Reply:
Thank you, this is wonderful.